test/rules.jl

```diff
@@ -31,8 +31,42 @@ _second(t) = Base.tuple_type_head(Base.tuple_type_tail(t))
 
     frx, fr = frule(cool, 1)
     @test frx == 2
-    @test fr(1) == 1
+    @test fr(NamedTuple(), 1) == (1,)
```

Comment:
If I understand correctly, the first argument is just ignored. Should we just make it convention to pass an empty tuple here instead, if we don't need this? `NamedTuple()` seems a bit long and potentially confusing to me.

Reply:
Fair point, for context:

If ever it is not ignored, because it has been overloaded,
because the thing is a functor,
then it will in the real world always be a `NamedTuple`, but with some fields.

I can see the arguement to use `()` for the case we know it will be ignored.
Someone else want to chime in,

We probably do not want to use this "shorthand".
```
julia> (;()...)
NamedTuple()
```